### DISCUSSION TITLE
Installing a base directory different from the par2 directory of the archive

### DISCUSSION THREAD

**Discussion by flarens** [Yutaka-Sawada/MultiPar#142] [Q&A] [answered]
In order not to clutter the main folder with content with service files, I want to put all par2 archives in a separate subfolder `_recovery`

If I use the following command for the content of the C:\test folder:
`par2j64.exe c -rf1 -rr10 -fu C:\test\_recovery\bd.par2 C:\test\_recovery\list.txt`
The base directory is automatically detected:
`Base Directory : "C:\test\_recovery\"`
And the program does not see the files:
`out of base-directory, C:\test\file1.zip`

The list of commands includes the `/d` command, which, judging by the description, should set the base directory. But it is absolutely unclear how to use it, there is a lack of working examples. I have to experiment for a long time without success.

I have tried the following options and they all give an error: `base-directory is invalid`

```
par2j64.exe c -rf1 -rr10 -fu -d C:\test C:\test\_recovery\bd.par2 C:\test\_recovery\list.txt
par2j64.exe c -rf1 -rr10 -fu -d C:\test\ C:\test\_recovery\bd.par2 C:\test\_recovery\list.txt
par2j64.exe c -rf1 -rr10 -fu -d C:\test\* C:\test\_recovery\bd.par2 C:\test\_recovery\list.txt
par2j64.exe c -rf1 -rr10 -fu -d C:\test\_recovery\bd.par2 C:\test C:\test\_recovery\list.txt
par2j64.exe c -rf1 -rr10 -fu -d C:\test\_recovery\bd.par2 C:\test\ C:\test\_recovery\list.txt
par2j64.exe c -rf1 -rr10 -fu -d C:\test\_recovery\bd.par2 C:\test\* C:\test\_recovery\list.txt
```

**Comment by Yutaka-Sawada** [comment, marked as answer]
> But it is absolutely unclear how to use it, there is a lack of working examples.

I'm not sure why some users don't read manual. I wrote some examples in `Command_par2j.txt`. Please refer section about `/d` option.

To distinguish options from filenames, don't put space between an option and it's related path. Because the rule is confusing, I wrote exact sample. Set base-directory like `-dC:\test`. If the path includes space, set like `-d"C:\test now"`.

**Reply by flarens** [reply]
Now it works perfectly.

The manual people read selectively what is required at a particular moment. Looking for the `/d` section and there is no mention that the command should be written inseparably with the address. I've never seen such syntax before and besides you yourself in the last question wrote that all options should be placed before all file names. In any case, a tiny line of the example `par2j64.exe c -rr10 -dC:\test C:\test\folder\bd.par2 file.zip` in the `/d` section would solve everything.

Thanks again for your help!

**Reply by Yutaka-Sawada** [reply]
> Looking for the /d section and there is no mention that the command should be written inseparably with the address. I've never seen such syntax before and besides you yourself in the last question wrote that all options should be placed before all file names.

Yes, the syntax is strange. I'm sorry for confusing newbie users. That is why I wrote example. I made this method to accept strange filenames, like `-d` or `/d`. Some users requested filenames starting with `-` or `/`. When your filename is `-d`, it's difficult to determine the word is intended as an option or a filename. To solve this problem, order and space are important meaning in the command-line.
